core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -383,59 +326,321 @@ func (l *logPollerWrapper) getCurrentCoordinators(ctx context.Context) (common.A
+				{Type: addressType}, // Client
+				{Type: uint64Type},  // SubscriptionId
+				{Type: uint32Type},  // CallbackGasLimit
+				{Type: uint72Type},  // AdminFee
+				{Type: uint72Type},  // DonFee
+				{Type: uint40Type},  // GasOverheadBeforeCallback
+				{Type: uint40Type},  // GasOverheadAfterCallback
+				{Type: uint32Type},  // TimeoutTimestamp
+				{Type: uint72Type},  // OperationFee
+			}
+			commitmentBytes, err := commitmentABI.Pack(
+				oracleRequest.Commitment.RequestId,
+				oracleRequest.Commitment.Coordinator,
+				oracleRequest.Commitment.EstimatedTotalCostJuels,
+				oracleRequest.Commitment.Client,
+				oracleRequest.Commitment.SubscriptionId,
+				oracleRequest.Commitment.CallbackGasLimit,
+				oracleRequest.Commitment.AdminFee,
+				oracleRequest.Commitment.DonFee,
+				oracleRequest.Commitment.GasOverheadBeforeCallback,
+				oracleRequest.Commitment.GasOverheadAfterCallback,
+				oracleRequest.Commitment.TimeoutTimestamp,
+				oracleRequest.Commitment.OperationFee,
+			)
+			if err != nil {
+				l.lggr.Errorw("LogsToRequests: failed to pack commitment bytes, skipping", err)
+			}
+
+			requests = append(requests, evmRelayTypes.OracleRequest{
+				RequestId:           oracleRequest.RequestId,
+				RequestingContract:  oracleRequest.RequestingContract,
+				RequestInitiator:    oracleRequest.RequestInitiator,
+				SubscriptionId:      oracleRequest.SubscriptionId,
+				SubscriptionOwner:   oracleRequest.SubscriptionOwner,
+				Data:                oracleRequest.Data,
+				DataVersion:         oracleRequest.DataVersion,
+				Flags:               oracleRequest.Flags,
+				CallbackGasLimit:    oracleRequest.CallbackGasLimit,
+				TxHash:              oracleRequest.Raw.TxHash,
+				OnchainMetadata:     commitmentBytes,
+				CoordinatorContract: coordinator.address,
+			})
+		}
+		return requests, nil
+	}
+
+	return nil, errors.Errorf("LogsToRequests: Unsupported Coordinator version %s", coordinator.typeAndVersion)
+}
+
+func (l *logPollerWrapper) logsToResponses(coordinator coordinator, responseLogs []logpoller.Log) ([]evmRelayTypes.OracleResponse, error) {
+	var responses []evmRelayTypes.OracleResponse
+
+	if strings.Contains(coordinator.typeAndVersion, "Functions Coordinator v1") {
+		parsingContract, err := functions_coordinator_1_1_0.NewFunctionsCoordinator110(coordinator.address, l.client)
+		if err != nil {
+			return nil, errors.Errorf("LogsToResponses: creating a contract instance for parsing failed")
+		}
+		for _, log := range responseLogs {
+			gethLog := log.ToGethLog()
+			oracleResponse, err := parsingContract.ParseOracleResponse(gethLog)
+			if err != nil {
+				l.lggr.Errorw("LogsToResponses: failed to parse a response log, skipping")
+				continue
+			}
+			responses = append(responses, evmRelayTypes.OracleResponse{
+				RequestId: oracleResponse.RequestId,
+			})
+		}
+		return responses, nil
+	}
+
+	if strings.Contains(coordinator.typeAndVersion, "Functions Coordinator v2") {
```

Comment:
Just reducing all these `if strings.Contains...` statements alone would probably have a measurable impact on code duplication.

Reply:
Moved some things around to mitigate the SonarQube code duplication. Take another look